docs/requirements.txt

```diff
@@ -0,0 +1,272 @@
+#
+# This file is autogenerated by pip-compile with python 3.10
+# To update, run:
+#
+#    pip-compile --allow-unsafe --generate-hashes --output-file=docs/requirements.txt --strip-extras docs/requirements.in
```

Comment:
@abhinavsingh note that this file is auto-generated and is checked in for reproducibility purposes.

Reply:
Thank you, I'll spend sometime to try this out.  I haven't used this option before, but I see where it's coming from.  I have now about a dozen TODOs on my list, thanks to you :D